modules/auxiliary/scanner/rsync/modules_list.rb

```diff
@@ -6,64 +6,201 @@
 require 'msf/core'
 
 class Metasploit3 < Msf::Auxiliary
-
   include Msf::Exploit::Remote::Tcp
   include Msf::Auxiliary::Scanner
   include Msf::Auxiliary::Report
 
+  RSYNC_HEADER = '@RSYNCD:'
+
   def initialize
     super(
-      'Name'        => 'Rsync Unauthenticated List Command',
-      'Description' => 'List all (listable) modules from a rsync daemon',
-      'Author'      => 'ikkini',
+      'Name'        => 'List Rsync Modules',
+      'Description' => %q(
+        An rsync module is essentially a directory share.  These modules can
+        optionally be protected by a password.  This module connects to and
+        negotiates with an rsync server, lists the available modules and,
+        optionally, determines if the module requires a password to access.
+      ),
+      'Author'      => [
+        'ikkini', # original metasploit module
+        'Jon Hart <jon_hart[at]rapid7.com>', # improved metasploit module
+        'Nixawk' # improved metasploit module
+      ],
       'References'  =>
         [
           ['URL', 'http://rsync.samba.org/ftp/rsync/rsync.html']
         ],
       'License'     => MSF_LICENSE
     )
+
     register_options(
       [
+        OptBool.new('TEST_AUTHENTICATION',
+                    [ true, 'Test if the rsync module requires authentication', true ]),
         Opt::RPORT(873)
-      ], self.class)
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SHOW_MOTD',
+                    [ true, 'Show the rsync motd, if found', false ]),
+        OptInt.new('READ_TIMEOUT', [ true, 'Seconds to wait while reading rsync responses', 2 ])
+      ]
+    )
+  end
+
+  def peer
+    "#{rhost}:#{rport}"
+  end
+
+  def read_timeout
+    datastore['READ_TIMEOUT']
+  end
+
+  def rsync_requires_auth?(rmodule)
```

Comment:
Good point.  That method had changed over time and what you suggested works for me.  Done.